### PR TITLE
Support test notification

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/send_test_notification.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+    platform_details["platform_info"] = {}
+    platform_details["platform_info"]["build_context"] = build_context
+    platform_details["platform_info"]["dlc_images"] = get_dlc_images(build_context)
+    platform_details["platform_info"]["test_type"] = os.getenv("TEST_TYPE")
+    platform_details["platform_info"]["codebuild_name"] = codebuild_name
+    platform_details["platform_info"]["codebuild_id"] = get_codebuild_project_id()
+    platform_details["platform_info"]["codebuild_url"] = get_target_url(codebuild_name)
+    platform_details["platform_info"]["cloudwatch_logs_url"] = get_cloudwatch_url(codebuild_name)
+    if build_context == "PR":
```
Comment (ohadkatz) [inline src/send_test_notification.py:78]: Might want to add these conditionals at the top, with an else statement raising an error that there is no build context.

File: src/send_test_notification.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+
+def parse_pytest_data():
+    """
+    Parse pytest output to get test results.
+    """
+    pytest_raw_data = get_pytest_output()
+    pytest_parsed_output = []
+
+    for f in pytest_raw_data:
```
Comment (ohadkatz) [inline src/send_test_notification.py:119]: Nit: Might make it more readable to use a better variable for the iterations

File: test/dlc_tests/conftest.py [hunk 3]
```diff
@@ -1241,6 +1242,15 @@ def pytest_runtest_setup(item):
     """
     # Handle quick check tests
     quick_checks_opts = [mark for mark in item.iter_markers(name="quick_checks")]
+
```
Comment (ohadkatz) [inline test/dlc_tests/conftest.py:1245]: Just make sure to uncomment this when ready, or remove if not needed

File: src/send_test_notification.py (hunk 1 above)
Comment (tejaschumbalkar) [reply src/send_test_notification.py:78]: done

File: src/send_test_notification.py (hunk 2 above)
Comment (tejaschumbalkar) [reply src/send_test_notification.py:119]: done

File: test/dlc_tests/conftest.py (hunk 3 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/conftest.py:1245]: done

**Review by ohadkatz — COMMENTED** [review]
LGTM, just remember to enable the notification system once ready. I can approve at that point once ready to go live 👍

File: testspec.yml [hunk 4]
```diff
@@ -27,6 +27,8 @@ phases:
   post_build:
     commands:
       - python src/send_status.py --status $CODEBUILD_BUILD_SUCCEEDING
+      # TODO: Enable test notification once ready
```
Comment (tejaschumbalkar) [inline testspec.yml:30]: Will create a separate PR to enable uncomment this, as a mark to making the feature live

**Review by ohadkatz — APPROVED** [review]

**Pull request merged**